realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -368,12 +368,9 @@ public void close() {
             realmNotifier.close();
         }
         synchronized (context) {
-            if (nativePtr != 0) {
-                nativeCloseSharedRealm(nativePtr);
-                // It is OK to clear the nativePtr. It has been saved to the NativeObjectReference when adding to the
-                // context.
-                nativePtr = 0;
-            }
+            nativeCloseSharedRealm(nativePtr);
+            // Don't reset the nativePtr since we still rely on Object Store to check if the given SharedRealm ptr
+            // is closed or not.
```

Comment:
`SharedRealm.isClosed()` should be updated as well.
No need to check `nativePtr` is null.

```
     public boolean isClosed() {
         return nativePtr == 0 || nativeIsClosed(nativePtr);
     }
 ```